File: Simple Coffee Machine/CoffeeMachine.java
import java.util.Scanner;

public class CoffeeMachine {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        int water = 400;
        int milk = 540;
        int coffeeBeans = 120;
        int cups = 9;
        int money = 550;
        System.out.println("Write action (buy, fill, take, remaining, exit):");
        String input = sc.next();
        System.out.println();

        while (!input.equals("exit")) {
            switch (input) {
                case "buy":
                    System.out.println("What do you want to buy? 1 - espresso, 2 - latte, 3 - cappuccino, back - to main menu:");
                    String choice = sc.next();
                    switch (choice) {
                        case "1":
                            if (water >= 250 && coffeeBeans >= 16 && cups >= 1) {
                                System.out.println("I have enough resources, making you a coffee!");
                                System.out.println();
                                water -= 250;
                                coffeeBeans -= 16;
                                cups -= 1;
                                money += 4;
                            } else {
                                if (water < 250) {
                                    System.out.println("Sorry, not enough water!");
                                } else if (coffeeBeans < 16) {
                                    System.out.println("Sorry, not enough coffee beans!");
                                } else if (cups < 1) {
                                    System.out.println("Sorry, not enough cups!");
                                }
                                System.out.println();
                            }
                            break;
                        case "2":
                            if (water >= 350 && milk >= 75 && coffeeBeans >= 20 && cups >= 1) {
                                System.out.println("I have enough resources, making you a coffee!");
                                System.out.println();
                                water -= 350;
                                milk -= 75;
                                coffeeBeans -= 20;
                                cups -= 1;
                                money += 7;
                            } else {
                                if (water < 350) {
                                    System.out.println("Sorry, not enough water!");
                                } else if (milk < 75) {
                                    System.out.println("Sorry, not enough milk!");
                                } else if (coffeeBeans < 20) {
                                    System.out.println("Sorry, not enough coffee beans!");
                                } else if (cups < 1) {
                                    System.out.println("Sorry, not enough cups!");
                                }
                                System.out.println();
                            }
                            break;
                        case "3":
                            if (water >= 200 && milk >= 100 && coffeeBeans >= 12 && cups >= 1) {
                                System.out.println("I have enough resources, making you a coffee!");
                                System.out.println();
                                water -= 200;
                                milk -= 100;
                                coffeeBeans -= 12;
                                cups -= 1;
                                money += 6;
                            } else {
                                if (water < 200) {
                                    System.out.println("Sorry, not enough water!");
                                } else if (milk < 100) {
                                    System.out.println("Sorry, not enough milk!");
                                } else if (coffeeBeans < 12) {
                                    System.out.println("Sorry, not enough coffee beans!");
                                } else if (cups < 1) {
                                    System.out.println("Sorry, not enough cups!");
                                }
                                System.out.println();
                            }
                            break;
                        case "back":
                            break;
                    }
                    break;
                case "fill":
                    System.out.println("Write how many ml of water you want to add:");
                    water += sc.nextInt();
                    System.out.println("Write how many ml of milk you want to add:");
                    milk += sc.nextInt();
                    System.out.println("Write how many grams of coffee beans you want to add:");
                    coffeeBeans += sc.nextInt();
                    System.out.println("Write how many disposable cups of coffee you want to add:");
                    cups += sc.nextInt();
                    System.out.println();
                    break;
                case "take":
                    System.out.println("I gave you $" + money);
                    money -= money;
                    System.out.println();
                    break;
                case "remaining":
                    System.out.println("The coffee machine has:");
                    System.out.println(water + " ml of water");
                    System.out.println(milk + " ml of milk");
                    System.out.println(coffeeBeans + " g of coffee beans");
                    System.out.println(cups + " disposable cups");
                    System.out.println("$" + money + " of money");
                    System.out.println();
                    break;
                case "exit":
                    input = "exit";
                    break;
            }
            if (!input.equals("exit")) {
                System.out.println("Write action (buy, fill, take, remaining, exit):");
                input = sc.next();
                System.out.println();
            }
        }

    }
}

File: Simple Coffee Machine/coffee_machine.py
water = 400
milk = 540
coffee_beans = 120
cups = 9
money = 550

print('Write action (buy, fill, take, remaining, exit):')
n = input()
print()

while n != 'exit':
    if n == 'buy':
        print('What do you want to buy? 1 - espresso, 2 - latte, 3 - cappuccino, back - to main menu:')
        x = input()
        if x == '1' and water >= 250 and coffee_beans >= 16 and cups >= 1:
            print('I have enough resources, making you a coffee!')
            water -= 250
            coffee_beans -= 16
            cups -= 1
            money += 4
        elif x == '2' and water >= 350 and milk >= 75 and coffee_beans >= 20 and cups >= 1:
            print('I have enough resources, making you a coffee!')
            water -= 350
            milk -= 75
            coffee_beans -= 20
            cups -= 1
            money += 7
        elif x == '3' and water >= 200 and milk >= 100 and coffee_beans >= 12 and cups >= 1:
            print('I have enough resources, making you a coffee!')
            water -= 200
            milk -= 100
            coffee_beans -= 12
            cups -= 1
            money += 6
        elif x == 'back':
            print()
        else:
            print('Sorry, not enough water!')
    elif n == 'fill':
        print('Write how many ml of water you want to add:')
        water += int(input())
        print('Write how many ml of milk you want to add:')
        milk += int(input())
        print('Write how many grams of coffee beans you want to add:')
        coffee_beans += int(input())
        print('Write how many disposable coffee cups you want to add:')
        cups += int(input())
    elif n == 'take':
        print(f'I gave you ${money}')
        money -= money
    elif n == 'remaining':
        print(f'The coffee machine has:\n{water} of water\n{milk} of milk\n{coffee_beans} of coffee beans\n{cups} of '
              f'disposable cups\n{money} of money')

    print()
    print('Write action (buy, fill, take, remaining, exit):')
    n = input()
    print()
